vmf/scripts/mods/vmf/modules/core/network.lua

```diff
@@ -289,7 +293,48 @@ local function vmf_network_recv(sender, channel_id, rpc_data1, rpc_data2)
         end
       end
     end)
+end
 
+local expected_pong_data_blocks = {}
+local partial_pong_mod_data = {}
+local partial_pong_rpc_data = {}
+
+local function vmf_network_recv(sender, channel_id, rpc_data1, rpc_data2)
+  if not _network_module_is_initialized then
+    return
+  end
+
+  if channel_id == RPC_VMF_REQUEST_CHANNEL_ID then -- rpc_vmf_request
+
+    network_debug("ping", "received", sender)
+
+    send_rpc_vmf_pong(sender)
+
+  elseif channel_id == RPC_VMF_RESPONCE_CHANNEL_ID then -- rpc_vmf_responce
+    -- @TODO: maybe I should protect it from sending by the player who's not in the game?
+
+	if rpc_data1 == "blocks" then
+		expected_pong_data_blocks[sender] = tonumber(rpc_data2)
+		partial_pong_mod_data[sender] = ""
+		partial_pong_rpc_data[sender] = ""
+	elseif expected_pong_data_blocks[sender] and expected_pong_data_blocks[sender] > 0 then
+		expected_pong_data_blocks[sender] = expected_pong_data_blocks[sender] - 1
+		if rpc_data1 then
+			partial_pong_mod_data[sender] = partial_pong_mod_data[sender] .. rpc_data1
+		end
+		if rpc_data2 then
```

Comment:
These will never be nil: `string.sub` returns an empty string if the interval is empty.

Reply:
Removed the `if`s. I also looked at http://www.lua.org/source/5.1/lvm.c.html#luaV_concat to verify that Lua optimizes the concatenation when the 2nd parameter is empty. Interestingly, it does not do the same if the 1st parameter is empty, but it shouldn't be a big deal.